### DISCUSSION TITLE
Does there have any way use this in server side?

### DISCUSSION THREAD

**Discussion by www-xu** [openapi-ts/openapi-typescript#1926] [Q&A]
**Description**

I'm working on a website, using clerk as the user management. I wanna inject the auth token provided by the clerk by middleware. But the method used to get the token can only be used within the Components or server side. Does there have a good way to inject token?

**Proposal**

I wanna inject the auth token by middleware without wrapping all methods or manually injecting it within Components.
Thanks

**Comment by drwpow** [comment]
Converted to a discussion because this sounds like more of a question about Clerk specifically, which I haven’t used, and isn’t related to this library. Could you provide more details of why the [middleware auth examples](https://openapi-ts.dev/openapi-fetch/middleware-auth) don’t work with Clerk (citing specific examples in their documentation)?

**Reply by www-xu** [reply]
thanks for your reply.

here is my code
```
const authMiddleware: Middleware = {
	async onRequest({request}) {
		const {isSignedIn, userId, getToken} = useAuth();


		if (userId) {
			request.headers.set("Authorization", `Bearer ${await getToken({template: 'Server'})}`);
		}

		return request;
	},
}
```

I tried to invoke `useAuth()` provided by the clerk. of course, it doesn't work, because useAuth can only be used within the Components.

the clerk also provides the `Auth()` method used on the server side, which works. but I have to wrap each method to export function. 

```
'use server'

import createClient, {Middleware} from "openapi-fetch";
import {paths} from "@/app/clients/server";
import {auth} from "@clerk/nextjs/server";

const authMiddleware: Middleware = {
	async onRequest({request}) {
		const {userId, getToken} = auth()

		if (userId) {
			request.headers.set("Authorization", `Bearer ${await getToken({template: 'Server'})}`);
		}

		return request;
	},
}

const server_client = createClient<paths>({baseUrl: process.env.URL});

server_client.use(authMiddleware);

export const pagingDictionary = async (page: number, size: number) => {
 	const {data, error} = await server_client.GET('/dictionary', {
 		params: {
 			query: {page: page, size: size},
 		}
 	})
 	return {data, error}
}

```

So, I wanna consult if there has any convenient way to wrap `GET`, `POST`, etc. on the server side.
